### PR TITLE
[Mobile] Buttons Block: Adds debounce in the add button action

### PR DESCRIPTION
**Pull request by antonis**
**Fixes** https://github.com/wordpress-mobile/gutenberg-mobile/issues/2238

`Gutenberg Mobile PR` -> https://github.com/wordpress-mobile/gutenberg-mobile/pull/2625

### Behavior without the fix
**iOS:** When user taps fast multiple times on side "+" icon to add multiple buttons the app falls in a loop
**Android:** When user taps fast multiple times on side "+" icon to add multiple buttons the app temporary freeze and the buttons are eventually created.
### Fix
A debounce has been added in the add button action.
Extending the fix to the `ButtonBlockAppender` was considered but the reported bug was not reproduced in any other block using it. Thus adding a delay on user action was not considered necessary.

## How has this been tested?

- Create a post or page
- Add a **buttons** block
- Tap fast multiple times on side "+" icon to add multiple buttons.
- Confirm that the app does not fall in a loop (iOS) or freezes (Android)

## Screenshots

|Without the fix|With the fix|
|---|---|
||

## Types of changes
Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/buttons/edit.native.js [hunk 1]
```diff
@@ -45,11 +46,13 @@ function ButtonsEdit( {
 		}
 	}, [ sizes ] );
 
+	const debounceAddNextButton = debounce( onAddNextButton, 500 );
```
Comment (lukewalczak) [inline packages/block-library/src/buttons/edit.native.js:49]: I have the impression that the value of `500` is a bit too big and wondering if `200` won't be enough? Could you please check that?

Comment (antonis) [reply packages/block-library/src/buttons/edit.native.js]: I was able to reproduce this up to 170ms. 200 seems quite reasonable.
I updated the PR

Comment (lukewalczak) [reply packages/block-library/src/buttons/edit.native.js]: Thanks!

**Review by lukewalczak — APPROVED** [review]
Thanks for the PR, it works so much better!

**Comment by pinarol** [comment]
Great fix! 🎉

**Pull request merged**